lib/src/configuration.rs

```diff
@@ -0,0 +1,53 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
+
+pub type BasicAuth = (String, Option<String>);
+
+#[derive(Debug, Clone)]
+pub struct ApiKey {
+    pub prefix: Option<String>,
+    pub key: String,
+}
+
+impl Configuration {
+    pub fn new(
+        base_path: String,
+        user_agent: Option<String>,
+        oauth_access_token: Option<String>,
```

Comment:
Why can't we set the other members of the struct, like `basic_auth` when creating a new config instance?

Reply:
I have added basic_auth in the constructor. @pavelnikonorov do you think we should add any other parameter to constructor?